diff --git a/boards/xtensa/esp32s3/esp32s3-devkit/src/Make.defs b/boards/xtensa/esp32s3/esp32s3-devkit/src/Make.defs
index 03c4fafc4f1..6c5b14d7135 100644
--- a/boards/xtensa/esp32s3/esp32s3-devkit/src/Make.defs
+++ b/boards/xtensa/esp32s3/esp32s3-devkit/src/Make.defs
@@ -59,6 +59,10 @@ ifeq ($(CONFIG_RPMSG),y)
 CSRCS += esp32s3_rpmsg.c
 endif
 
+ifeq ($(CONFIG_USBDEV_COMPOSITE),y)
+CSRCS += esp32s3_composite.c
+endif
+
 DEPPATH += --dep-path board
 VPATH += :board
 CFLAGS += ${INCDIR_PREFIX}$(TOPDIR)$(DELIM)arch$(DELIM)$(CONFIG_ARCH)$(DELIM)src$(DELIM)board$(DELIM)board
@@ -66,3 +70,7 @@ CFLAGS += ${INCDIR_PREFIX}$(TOPDIR)$(DELIM)arch$(DELIM)$(CONFIG_ARCH)$(DELIM)src
 ifeq ($(CONFIG_ETC_ROMFS),y)
 RCSRCS  = etc/init.d/rc.sysinit etc/init.d/rcS
 endif
+
+ifeq ($(CONFIG_LCD_ST7789),y)
+CSRCS += esp32s3_st7789.c
+endif
diff --git a/boards/xtensa/esp32s3/common/scripts/esp32s3_rom_aliases.ld b/boards/xtensa/esp32s3/common/scripts/esp32s3_rom_aliases.ld
index 27d70ec783e..f7431d8c1b6 100644
--- a/boards/xtensa/esp32s3/common/scripts/esp32s3_rom_aliases.ld
+++ b/boards/xtensa/esp32s3/common/scripts/esp32s3_rom_aliases.ld
@@ -26,6 +26,7 @@
 
 PROVIDE( cache_dbus_mmu_set = Cache_Dbus_MMU_Set );
 PROVIDE( cache_ibus_mmu_set = Cache_Ibus_MMU_Set );
+Cache_Invalidate_Addr = 0x400016b0;
 PROVIDE( cache_invalidate_addr = Cache_Invalidate_Addr );
 PROVIDE( cache_invalidate_dcache_all = Cache_Invalidate_DCache_All );
 PROVIDE( cache_invalidate_icache_all = Cache_Invalidate_ICache_All );
diff --git a/drivers/syslog/Make.defs b/drivers/syslog/Make.defs
index e25107090a4..a3483a0c4a5 100644
--- a/drivers/syslog/Make.defs
+++ b/drivers/syslog/Make.defs
@@ -39,7 +39,7 @@ endif
 ifeq ($(CONFIG_RAMLOG),y)
   CSRCS += ramlog.c
 
-  ifneq ($(CONFIG_RAMLOG_BUFFER_SECTION),"")
+  ifneq ($(CONFIG_RAMLOG_BUFFER_SECTION),)
     CFLAGS += ${DEFINE_PREFIX}RAMLOG_BUFFER_SECTION
   endif
 endif
