diff --git a/drivers/input/ft5x06.c b/drivers/input/ft5x06.c
index be5db167865..36b99546c60 100644
--- a/drivers/input/ft5x06.c
+++ b/drivers/input/ft5x06.c
@@ -39,6 +39,7 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
+#include <syslog.h>
 
 #include <sys/types.h>
 
@@ -305,8 +306,6 @@ static void ft5x06_data_worker(FAR void *arg)
 
       if (sample->tdstatus > 0 && sample->tdstatus <= FT5X06_MAX_TOUCHES)
         {
-          /* Keep it at the minimum if touches are detected. */
-
           priv->delay = POLL_MINDELAY;
         }
       else if (priv->delay < POLL_MAXDELAY)
@@ -508,8 +507,7 @@ static ssize_t ft5x06_sample(FAR struct ft5x06_dev_s *priv, FAR char *buffer,
   point             = sample->point;
   point[0].id       = id;
   point[0].flags    = g_event_map[event];
-  point[0].x        = x;
-  point[0].y        = y;
+  { int16_t _dx=y, _dy=320-x; if(_dx<0)_dx=0; else if(_dx>479)_dx=479; if(_dy<0)_dy=0; else if(_dy>319)_dy=319; point[0].x=_dx; point[0].y=_dy; }
   point[0].h        = 0;
   point[0].w        = 0;
   point[0].pressure = 0;
@@ -945,6 +943,14 @@ static int ft5x06_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
         }
         break;
 
+      case TSIOC_GETMAXPOINTS:  /* arg: Pointer to uint8_t max touch points */
+        {
+          FAR uint8_t *ptr = (FAR uint8_t *)((uintptr_t)arg);
+          DEBUGASSERT(ptr != NULL);
+          *ptr = 1;
+        }
+        break;
+
       default:
         ret = -ENOTTY;
         break;
